blaze/mahimahi/server/server.py

```diff
@@ -139,7 +146,7 @@ def start_server(
                     if extract_critical_requests:
                         # if this is an html file, then we want to insert our snippet here
```

Comment:
Make everything from line 147-181 a new function that takes in a file and returns a body with the injected script (if it is able to do so). Also remove all prints and use `log`. Something like this:

```
def inject_extract_critical_requests_javascript(file):
    # Does what it is supposed to do, but throws an exception if it cannot do it

# ...

body = file.body
if extract_critical_requests and "text/html" in file.headers.get("content-type", ""):
    try:
        body = inject_extract_critical_requests_javascript(file)
    except (TypeError, UnicodeDecodeError, ValueError) as err:
        log.warn("failed to inject javascript", file=file.uri, err=str(err))

# write `body` once here
# ....

```